### DISCUSSION TITLE
Deploying to Heroku

### DISCUSSION THREAD

**Discussion by dfaruque** [serenity-is/Serenity#6469] [Help] [answered]
I have tried to deploy a serenity .NET 6 app to Heroku using the following build pack.
https://github.com/jincod/dotnetcore-buildpack

but while deploying to heroku it fails with the following log:
```
Microsoft (R) Build Engine version 17.2.0+41abc5629 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.
  Determining projects to restore...
  Restored /tmp/build_ca403de6/xxx.Web.csproj (in 10.28 sec).

  Installing NPM packages...
  /usr/bin/sh: 2: /tmp/tmpb274bb66ba0a4800a4bb669cf937bf3c.exec.cmd: npm: not found
/tmp/build_ca403de6/xxx.Web.csproj(56,5): error MSB3073: The command "npm install" exited with code 127.
 !     Push rejected, failed to compile ASP.NET Core app.
 !     Push failed
```

Anyone have the experience to deploy a serenity app to heroku?

**Comment by edwardch** [comment, marked as answer]
I haven't tried .NET or Serenity on Heroku personally, but seems like you need to specify different build packs: https://devcenter.heroku.com/articles/using-multiple-buildpacks-for-an-app

I hope that helps in some way...

**Reply by dfaruque** [reply]
Thank you. I have resolved the issue using the following build-packs together
![image](https://user-images.githubusercontent.com/16621563/181776502-579b4881-6c49-45f1-ae80-f810777a346e.png)
